REFERENCE.md

```diff
@@ -0,0 +1,310 @@
+
+##### `name`
+
+namevar
+
+An arbitrary name used as the identity of the resource.
+
+##### `section`
+
+
+
+Default value: ''
+
+##### `setting`
+
+The name of the setting to be defined.
+
+##### `force_new_section_creation`
+
+Valid values: `true`, `false`, yes, no
+
+Create setting only if the section exists
+
+Default value: `true`
+
+##### `path`
+
+The ini file Puppet will ensure contains the specified setting.
+
+##### `show_diff`
+
+Valid values: `true`, md5, `false`
+
+Whether to display differences when the setting changes.
+
+Default value: `true`
+
+##### `key_val_separator`
+
+
+
+Default value:  = 
+
+##### `section_prefix`
+
+
+
+Default value: [
+
+##### `section_suffix`
+
+
+
+Default value: ]
+
+##### `indent_char`
+
+
+
+Default value:  
+
+##### `indent_width`
+
+
+
+##### `refreshonly`
+
+Valid values: `true`, `false`, yes, no
+
+
+
+Default value: `false`
+
+### ini_subsetting
+
+An arbitrary name used as the identity of the resource.
+
+#### Properties
+
+The following properties are available in the `ini_subsetting` type.
+
+##### `ensure`
+
+Valid values: present, absent
+
+The basic property that the resource should be in.
+
+Default value: present
+
+##### `value`
+
+The value of the subsetting to be defined.
+
+#### Parameters
+
+The following parameters are available in the `ini_subsetting` type.
+
+##### `name`
+
+namevar
+
+An arbitrary name used as the identity of the resource.
+
+##### `section`
+
+
+
+Default value: ''
+
+##### `setting`
+
+The name of the setting to be defined.
+
+##### `subsetting`
+
+The name of the subsetting to be defined.
+
+##### `subsetting_separator`
+
+The separator string between subsettings. Defaults to " "
+
+Default value:  
+
+##### `subsetting_key_val_separator`
+
+The separator string between the subsetting name and its value. Defaults to the empty string.
+
+Default value: ''
+
+##### `path`
+
+The ini file Puppet will ensure contains the specified setting.
+
+##### `show_diff`
+
+Valid values: `true`, md5, `false`
+
+Whether to display differences when the setting changes.
+
+Default value: `true`
+
+##### `key_val_separator`
+
+
+
+Default value:  = 
+
+##### `quote_char`
+
+
+
+Default value: ''
+
+##### `use_exact_match`
+
+Valid values: `true`, `false`
+
+Set to true if your subsettings don\'t have values and you want to use exact matches to determine if the subsetting exists. See MODULES-2212
+
+Default value: `false`
+
+##### `insert_type`
+
+Valid values: start, end, before, after, index
+
+Where the new subsetting item should be inserted?
+
+* :start  - insert at the beginning of the line.
+* :end    - insert at the end of the line (default).
+* :before - insert before the specified element if possible.
+* :after  - insert after the specified element if possible.
+* :index  - insert at the specified index number.
+
+Default value: end
+
+##### `insert_value`
+
+The value for the insert types which require one.
+
+## Functions
+
+### create_ini_settings
```

Comment:
Function not using Puppet Strings syntax. https://github.com/puppetlabs/puppetlabs-accounts/blob/master/lib/puppet/functions/accounts_ssh_options_parser.rb might help to figure out how to do this.